ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala

```diff
@@ -306,6 +311,44 @@ class ContractsService(
     }
   }
 
+  def refreshCache(
+      jwt: Jwt,
+      jwtPayload: JwtPayload,
+      refreshCacheRequest: RefreshCacheRequest,
+  )(implicit
+      lc: LoggingContextOf[InstanceUUID with RequestID],
+      metrics: HttpJsonApiMetrics,
+  ): Future[SearchResult[Error \/ domain.RefreshCacheResult]] = {
+    val ledgerId = toLedgerId(jwtPayload.ledgerId)
+    val optOffsetToUpdate = refreshCacheRequest.offset
+    getTermination(jwt, ledgerId)(lc).map { optLedgerEnd =>
+      optLedgerEnd.cata(
+        { ledgerEnd =>
+          daoAndFetch.cata(
+            { case (dao, fetchService) =>
+              val response: Source[Error \/ domain.RefreshCacheResult, NotUsed] = {
+                val futureValue =
+                  dao
+                    .transact(
+                      fetchService.fetchAndRefreshCache(jwt, ledgerId, ledgerEnd, optOffsetToUpdate)
+                    )
+                    .unsafeToFuture()
```

Comment:
Please use `unsafeRunAsync` as suggested in
https://github.com/digital-asset/daml/pull/16987#pullrequestreview-1480429879

Reply:
Sorry, if you explained somewhere else I can't find it. Why can't you use that function here?

Reply:
Ah, found it. Fair enough.

https://github.com/digital-asset/daml/pull/16987#discussion_r1231841302